### PR TITLE
ENH: Updates required by updated ITK

### PR DESCRIPTION
**Pull request by Leengit**
ENH: Add .clang-format style file.
ENH: Use ITK_ prefix for macro names.
ENH: Bump to C++14.
ENH: Bump to CMake minimum 3.16.3.
ENH: Update itk-git-tag in build-test-package.yml.
STYLE: Apply clang-format for C++ and black for Python.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,151 @@
+## This config file is only relevant for clang-format version 8.0.0
```
Comment (dzenanz) [inline .clang-format:1]: This should be a `STYLE:` commit, not a `WIP:`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.10.2)
```
Comment (dzenanz) [inline CMakeLists.txt]: Also bump to 3.16.3

File: .clang-format (hunk 1 above)
Comment (dzenanz) [reply .clang-format:1]: Also, you could break it up into "adding style file" and "applying clang-format".

Comment (Leengit) [reply .clang-format:1]: I forgot to update that.  Will do.

File: CMakeLists.txt (hunk 2 above)
Comment (Leengit) [reply CMakeLists.txt]: Will do.

File: .clang-format (hunk 1 above)
Comment (Leengit) [reply .clang-format:1]: > Also, you could break it up into "adding style file" and "applying clang-format".

Will do.

File: test/itkStructurePreservingColorNormalizationFilterTest.cxx [hunk 3]
```diff
@@ -100,113 +101,118 @@ int itkStructurePreservingColorNormalizationFilterTest( int argc, char * argv[]
...
+  using ImageType = itk::Image<PixelType, Dimension>; // IRGBUC2
 
-  using FilterType = itk::StructurePreservingColorNormalizationFilter< ImageType >;
+  using FilterType = itk::StructurePreservingColorNormalizationFilter<ImageType>;
   FilterType::Pointer filter = FilterType::New();
   // filter->SetColorIndexSuppressedByHematoxylin( 0 );
   // filter->SetColorIndexSuppressedByEosin( 1 );
 
-  EXERCISE_BASIC_OBJECT_METHODS( filter, StructurePreservingColorNormalizationFilter, ImageToImageFilter );
+  EXERCISE_BASIC_OBJECT_METHODS(filter, StructurePreservingColorNormalizationFilter, ImageToImageFilter);
```
Comment (dzenanz) [inline test/itkStructurePreservingColorNormalizationFilterTest.cxx]: `ITK_` prefix

File: test/itkStructurePreservingColorNormalizationFilterTest.cxx [hunk 4]
```diff
@@ -100,113 +101,118 @@ int itkStructurePreservingColorNormalizationFilterTest( int argc, char * argv[]
...
   WriterType::Pointer writer = WriterType::New();
-  writer->SetFileName( outputImageFileName );
-  writer->SetInput( filter->GetOutput() );
-  writer->SetUseCompression( true );
+  writer->SetFileName(outputImageFileName);
+  writer->SetInput(filter->GetOutput());
+  writer->SetUseCompression(true);
 
-  TRY_EXPECT_NO_EXCEPTION( writer->Update() );
+  TRY_EXPECT_NO_EXCEPTION(writer->Update());
```
Comment (dzenanz) [inline test/itkStructurePreservingColorNormalizationFilterTest.cxx]: `ITK_` prefix

**Review by dzenanz — COMMENTED** [review]
I guess `ITK_` prefix got lost in history rewrite?

File: test/itkStructurePreservingColorNormalizationFilterTest.cxx (hunk 3 above)
Comment (Leengit) [reply test/itkStructurePreservingColorNormalizationFilterTest.cxx]: We don't call you "eagle eye" for nothing.  Thanks!

**Review by dzenanz — APPROVED** [review]

**Comment by Leengit** [comment]
The `StructurePreservingColorNormalizationKWStyleTest` is failing.  Maybe it is still looking for Whitesmiths formatting instead of clang_format.  I am not seeing how to update that.  @dzenanz do you know?

**Comment by dzenanz** [comment]
Template parameters
https://github.com/InsightSoftwareConsortium/ITKColorNormalization/blob/51558907bd5375f5eb8ffb39cb7fde6286f8d7c1/include/itkStructurePreservingColorNormalizationFilter.h#L296
don't have the expected names (e.g. `TScalar` or `VRows`).

**Comment by Leengit** [comment]
> Template parameters
> 
> https://github.com/InsightSoftwareConsortium/ITKColorNormalization/blob/51558907bd5375f5eb8ffb39cb7fde6286f8d7c1/include/itkStructurePreservingColorNormalizationFilter.h#L296
> 
> don't have the expected names (e.g. `TScalar` or `VRows`).

Thanks!  Pushed.

**Pull request merged**